tests/api/printf.cpp

```diff
@@ -153,29 +172,29 @@ TEST_F(WithCommandQueue, TooLongPrintf) {
     size_t lws = 1;
     EnqueueNDRangeKernel(kernel, 1, nullptr, &gws, &lws, 0, nullptr, nullptr);
     Finish();
-
-    releaseStdout(fd);
-    auto printf_buffer = getStdoutContent();
-    ASSERT_NE(printf_buffer, nullptr);
-
-    // We only get the first 2 prints because the buffer is too small to get the
-    // last one.
-    const char* message = "get_global_id(0) = 0\nget_global_id(1) = 0\n";
-    ASSERT_STREQ(printf_buffer, message);
+    // Reset the buffer if complete, otherwise keep the remaining part
+    ASSERT_STREQ(GLOBAL_BUFFER.data(), message);
+    BUFFER_FILL_LEVEL = 0;
+    GLOBAL_BUFFER.clear();
+    GLOBAL_BUFFER.shrink_to_fit();
```

Comment:
Should not be needed here. Everything could be done at the beginning. We can leave the buffer as it is at the end